pkg/config/config.go

```diff
@@ -36,6 +42,59 @@ type Install struct {
 	BindMounts             []string          `yaml:"bind_mounts,omitempty"`
 }
 
+func NewConfig(opts ...GenericOptions) *Config {
+	log := v1.NewLogger()
+
+	defaultPlatform, err := v1.NewPlatformFromArch(runtime.GOARCH)
```

Comment:
Can we name this something other than default? maybe underlyingPlatform or hostPlatform, or something like that?

Reply:
done!